src/mjolnir/pbfgraphparser.cc

```diff
@@ -374,19 +376,30 @@ struct graph_callback : public OSMPBF::Callback {
     // Add the refs to the reference list and mark the nodes that care about when processing nodes
     loop_nodes_.clear();
     for (size_t i = 0; i < nodes.size(); ++i) {
+      // If we've seen this node anywhere else then its an intersection (ie graph node)
       const auto& node = nodes[i];
       if (shape_.get(node)) {
         intersection_.set(node);
         ++osmdata_.edge_count;
       } else {
         ++osmdata_.node_count;
       }
-      way_nodes_->push_back({{node}, static_cast<uint32_t>(ways_->size()), static_cast<uint32_t>(i)});
       shape_.set(node);
+
+      // Check whether or not the node is a duplicate or a turn around
+      // A turn around is the node at which a way doubles back on itself eg. way=node_0,node_1,node_0
+      OSMNode osm_node{node};
+      auto inserted = loop_nodes_.insert(std::make_pair(node, i));
+      osm_node.duplicate_ =
+          !inserted.second || (i != 0 && i != nodes.size() - 1 && nodes[i - 1] == nodes[i + 1]);
```

Comment:
when parsing we mark any nodes as "duplicate" if they were already referenced by the same way or if they were a turn around (the point at which it doubles back)

Reply:
Nit for readability: can you pull out `(i != 0 && i != nodes.size() - 1 && nodes[i - 1] == nodes[i + 1])` into a variable and give it a meaningful name?